pkg/apis/v1alpha1/handlers/customers.go

```diff
@@ -0,0 +1,50 @@
+package handlers
```

Comment:
are we moving all business logic will reside in `pkg/apis`?

Reply:
Good question. Prefer in `pkg/handlers`

Reply:
lgtm

we did discuss this before, to proposed (slightly modified) structure is

```
.                                   #
├── Gopkg.toml                      #
├── Makefile                        #
├── README.md                       # 
├── apis                            # 存放 apidocs (swagger json)
│   ├── api.v1.json                 #
│   └── api.v2.json                 #
├── bin                             # 存放编译后的二进制文件
│   └── nirvana-myproject           #
├── build                           # 存放 Dockerfile
│   └── nirvana-myproject           #
│       └── Dockerfile              #
├── cmd                             # 存放项目的启动命令
│   └── nirvana-myproject           #
│       └── main.go                 #
├── nirvana.yaml                    #
├── pkg                             # 存放 api 需要用到的结构体与相应的 converters, 按版本区分
│   ├── apis                        #
│   │   └── v1                      #
│   │       ├── converters          #
│   │       │   └── converters.go   #
│   │       └── types.go            #
│   │       ├── descriptors          #
│   │           ├── descriptors.go      #
│   │           └── message.go          # 对应 message 业务逻辑的 API 定义
│   ├── filters                     # 存放 HTTP Request 过滤器
│   │   └── filter.go               #
│   ├── handler                     # 存放 API 需要用到的逻辑处理
│   │   └── message.go              #
│   ├── middlewares                 # 存放中间件
│   │   └── middlewares.go          #
│   ├── modifiers                   # 存放 Definition 修改器
│   │   └── modifiers.go            #
│   └── version                     # 项目版本信息目录
│       └── version.go              #
└── vendor                          #
```

Reply:
a unified project structure definitely makes it easier to read new projects

3.0 might be a good chance to reinforce this idea since many of us will be migrating our repos for COS

Reply:
Will enforce the unification of projects' structure in v3.0.

Reply:
1. `handler` or `handlers`?
2. Do we need to separate handler/middlewares/modifiers into different versions like `v1` and `v2`?

Reply:
> 1. `handler` or `handlers`?
> 2. Do we need to separate handler/middlewares/modifiers into different versions like `v1` and `v2`?

The same question that I want to ask. `handler` isn't something global like `middlewares` or `modifiers`, so maybe:

```
├── pkg                             
│   ├── apis                        
│   │   └── v1                      
│   │       ├── converters          
│   │       │   └── converters.go   
│   │       └── types.go            
│   │       ├── descriptors          
│   │           ├── descriptors.go      
│   │           └── message.go          
│   |       ├── handlers                     
│   │           └── message.go              
```

Reply:
I follow the structure of [nirvana-template-project](https://github.com/caicloud/nirvana-template-project/tree/master/pkg). Is this not the standard structure generated by `nirvana init`?

Reply:
> Is this not the standard structure generated by `nirvana init`?

Nope, see https://github.com/caicloud/nirvana-practice/pull/5#issuecomment-620364037. The question is do we need to discuss the structure again?

Reply:
Strictly speaking, handler/middlewares/modifiers should all be moved into `pkg/apis/v1` (e.g. you can't say for sure that modifiers won't touch any v1 data structure). 

But this makes the code hard to reason about.  I'd vote for current structure and keep it as is.  The `pkg/apis` is only used for user-facing structures and routes.

Reply:
> But this makes the code hard to reason about. I'd vote for current structure and keep it as is. The `pkg/apis` is only used for user-facing structures and routes.

Agree

Reply:
> 1. `handler` or `handlers`?
> 2. Do we need to separate handler/middlewares/modifiers into different versions like `v1` and `v2`?

1. `handler`
2. no separation inside handler/middlewares/modifiers

Agreed?

Reply:
I think we can move to #318 for more discussions instead of here.